dali/test/tensor_test_utils.h

```diff
@@ -298,6 +303,12 @@ uniform_distribution(T lo, T hi) {
   return std::uniform_real_distribution<T>(lo, hi);
 }
 
+template <typename T>
+auto uniform_distribution(float16 lo, float16 hi, std::enable_if_t<is_half<T>::value>* = nullptr) {
+  return [lo, hi](auto &g) {
+    return static_cast<float16>(std::uniform_real_distribution<float>(lo, hi)(g));
```

Comment:
I'm not sure if you can do that in general, as FP16 has different ranges compared to FP32, in this case you won't get true uniform_real_distribution. But for testing utility it should do.

Reply:
Reverted that, the FP16 is broken in a way that doesn't allow me to use it in this test.